game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/BlockObjectCreator.scala

```diff
@@ -52,4 +52,42 @@ class BlockObjectCreator() {
+
+    val blockWindow = 15
+    var blockNumber = 1 + blockWindow
+
+    do {
+      val blocksInWindow = blocks.filter(x => x.getPosition().getBlockNumber() >= blockNumber - blockWindow && x.getPosition().getBlockNumber() <= blockNumber)
+
+      val isMudToRemove = //goal is don't want solid block of mud over window
+        (blocksInWindow.exists(x => x.getPosition().getBlockNumber() >= blockNumber - blockWindow && x.getPosition().getBlockNumber() <= blockNumber && x.getPosition().getLane() == 1 && (x.getMapObject() == Config.MUD_MAP_OBJECT || x.getMapObject() == Config.WALL_MAP_OBJECT))) &&
+        (blocksInWindow.exists(x => x.getPosition().getBlockNumber() >= blockNumber - blockWindow && x.getPosition().getBlockNumber() <= blockNumber && x.getPosition().getLane() == 2 && (x.getMapObject() == Config.MUD_MAP_OBJECT || x.getMapObject() == Config.WALL_MAP_OBJECT))) &&
+        (blocksInWindow.exists(x => x.getPosition().getBlockNumber() >= blockNumber - blockWindow && x.getPosition().getBlockNumber() <= blockNumber && x.getPosition().getLane() == 3 && (x.getMapObject() == Config.MUD_MAP_OBJECT || x.getMapObject() == Config.WALL_MAP_OBJECT))) &&
+        (blocksInWindow.exists(x => x.getPosition().getBlockNumber() >= blockNumber - blockWindow && x.getPosition().getBlockNumber() <= blockNumber && x.getPosition().getLane() == 4 && (x.getMapObject() == Config.MUD_MAP_OBJECT || x.getMapObject() == Config.WALL_MAP_OBJECT)))
+
+      if(isMudToRemove) {
+        val blocksBehindWhereObstacleCouldBeRemoved = blocksInWindow.filter(x => x.getPosition().getBlockNumber() == blockNumber - blockWindow && (x.getMapObject() != Config.MUD_MAP_OBJECT && x.getMapObject() != Config.WALL_MAP_OBJECT))
+        if(blocksBehindWhereObstacleCouldBeRemoved.length > 0) {
+          var indexOfBlockBehindBlocksToRemoveMudFrom = 0
+          if(blocksBehindWhereObstacleCouldBeRemoved.length > 1) {
+            indexOfBlockBehindBlocksToRemoveMudFrom = randomNumberGenerator.nextInt(blocksBehindWhereObstacleCouldBeRemoved.length - 1)
+          }
+          val blockBehindWhereMudShouldBeRemoved = blocksBehindWhereObstacleCouldBeRemoved(indexOfBlockBehindBlocksToRemoveMudFrom)
+          val blockNumberOfInterest = blockBehindWhereMudShouldBeRemoved.getPosition().getBlockNumber()
+          val laneOfInterest = blockBehindWhereMudShouldBeRemoved.getPosition().getLane()
+          val blocksToRemoveMudFrom = blocks.filter(x => x.getPosition().getBlockNumber() >= blockNumberOfInterest && x.getPosition().getBlockNumber() <= blockNumberOfInterest + blockWindow && x.getPosition().getLane() == laneOfInterest)
+          blocksToRemoveMudFrom.foreach(x => {
+            if(x.getMapObject() == Config.MUD_MAP_OBJECT || x.getMapObject() == Config.WALL_MAP_OBJECT) {
+              x.setMapObject(Config.EMPTY_MAP_OBJECT)
+            }
+          })
+        }
+      }
+
+      blockNumber = blockNumber + 1
+    } while(blockNumber < Config.TRACK_LENGTH)
```

Comment:
This will only fix vertical lines, right? 

You could try out [convolution](https://en.wikipedia.org/wiki/Kernel_(image_processing)) for this instead, it will compare a pixel/block to it's immediate neighbors to decide if a block should stay or be removed, even if the neighbors are diagonal

Reply:
Nah it's over a window of blocks. The nett effect is that no matter what, there wall always be a path of atleast <blockwindow> blocks that is clear of mud and walls. 

The kernel is blockwindow x 4 (since we want to cover all lanes). Its a simplified version of your suggestion